homeassistant/components/pyload/sensor.py

```diff
@@ -77,16 +80,32 @@ def setup_platform(
     username = config.get(CONF_USERNAME)
     password = config.get(CONF_PASSWORD)
     monitored_types = config[CONF_MONITORED_VARIABLES]
-    url = f"{protocol}://{host}:{port}/api/"
+    url = f"{protocol}://{host}:{port}/"
 
+    session = async_create_clientsession(
+        hass,
+        verify_ssl=False,
+        cookie_jar=CookieJar(unsafe=True),
+    )
+    pyloadapi = PyLoadAPI(session, api_url=url, username=username, password=password)
     try:
-        pyloadapi = PyLoadAPI(api_url=url, username=username, password=password)
-    except (
-        requests.exceptions.ConnectionError,
-        requests.exceptions.HTTPError,
-    ) as conn_err:
-        _LOGGER.error("Error setting up pyLoad API: %s", conn_err)
-        return
+        await pyloadapi.login()
+    except CannotConnect as conn_err:
+        raise ConfigEntryNotReady(
+            translation_domain=DOMAIN,
+            translation_key="connection_exception",
+        ) from conn_err
+    except ParserError as e:
+        raise ConfigEntryNotReady(
+            translation_domain=DOMAIN,
+            translation_key="parse_exception",
+        ) from e
+    except InvalidAuth as e:
+        raise ConfigEntryAuthFailed(
```

Comment:
Since this isn't a config entry this won't work

`ConfigEntryAuthFailed` starts `async_step_reauth` in a config flow but there is no config flow here.

Reply:
yeah, sorry, config flow comes with another PR ^^ 
So the right exception here would also be `PlatformNotReady` ?